workspaces/src/operations.rs

```diff
@@ -513,30 +513,44 @@ impl TransactionStatus {
     /// is in an unexpected state. The error should have further context. Otherwise, if an
     /// `Ok` value with [`Poll::Pending`] is returned, then the transaction has not finished.
     pub async fn status(&self) -> Result<Poll<ExecutionFinalResult>> {
-        let result = self
+        let rpc_resp = self
             .worker
             .client()
             .tx_async_status(
                 &self.sender_id,
                 near_primitives::hash::CryptoHash(self.hash.0),
-                near_primitives::views::TxExecutionStatus::Final,
+                TxExecutionStatus::Included,
```

Comment:
Indeed, final is not needed, but looking at the implementation, it feels that you actually need ExecutedOptimistic since the function needs to return the transaction outcome, right?

Reply:
nah, this is just polling the transaction status, so ExecutedOptimistic would have the transaction already be slightly  completed. This function should just be returning immediately where ExecutedOptimistic would take some time to resolve

Reply:
Ah, I misinterpreted `Poll<>` part in the function signature. I thought about it as part of the native Rust futures polling, but it is not.